fuseimg/datasets/ukbb_neck_to_knee.py

```diff
@@ -210,11 +213,15 @@ def dataset(
         :param is_female                    filter only male/females from database
         :return: DatasetDefault object
         """
-        if is_female == None:
-            all_sample_ids = [file for file in os.listdir(data_dir) if '.zip' in file]
-        else:
-            all_sample_ids = [file for file in os.listdir(data_dir) if '.zip' in file]
-            all_sample_ids = list(set(input_source_gt[input_source_gt['is female'] == is_female][input_source_gt['file'].isin(all_sample_ids)]['file'].to_list()))
+
+        existing_files = [file for file in os.listdir(data_dir) if '.zip' in file]
+        existing_sample_id_fields = [f.split("_") for f in existing_files]
+        existing_sample_ids = set([a[0] + "_*_" + a[2] + "_" + a[3] for a in existing_sample_id_fields])
+        a_filter = input_source_gt['file'].isin(existing_sample_ids)
```

Comment:
not sure why split and then merge again from existing_files  to existing_sample_ids .
you wanted to use later existing_sample_id_fields  for other fields? 
input_source_gt['file'] should be in which format?

Reply:
Not sure I understand the question.
existing_files =  a list filenames that exist on the disk. This names are NOT sample_ids
existing_sample_ids = sample_ids matching the existing files. These are of the form <eid>_*_<visit>_0
input_source_gt['file'] contains sample_id == file pattern = <eid>_*_<visit>_0